lib/elixir/unicode/unicode.ex

```diff
@@ -403,3 +415,87 @@ defmodule String.Graphemes do
     {acc, string}
   end
 end
+
+defmodule String.Normalizer do
+  @moduledoc false
+
+  decomposition_path = Path.join(__DIR__, "Decomposition.txt")
+
+  decompositions = Enum.reduce File.stream!(decomposition_path), [], fn(line, acc) ->
+    [key, first, second, third, fourth, _] = :binary.split(line, ";", [:global])
+    decomposition =
+      String.Hex.to_binary(first) <> (String.Hex.to_binary(second) || "") <>
+                                     (String.Hex.to_binary(third)  || "") <>
+                                     (String.Hex.to_binary(fourth) || "")
+    [{String.Hex.to_binary(key), decomposition} | acc]
+  end
+
+  combining_class_path = Path.join(__DIR__, "CombiningClasses.txt")
+
+  combining_classes = Enum.reduce File.stream!(combining_class_path), [], fn(line, acc) ->
+    [codepoint, class, _] = :binary.split(line, ";", [:global])
+    [{String.to_integer(codepoint, 16), class} | acc]
+  end
+
+  # Compare
+
+  def equivalent?(string, another_string) do
+    normalize(string, :nfd) == normalize(another_string, :nfd)
+  end
+
+  # Normalize
+
+  def normalize(string, :nfd) when is_binary(string) do
+    normalize_nfd(string, "")
+  end
+
+  defp normalize_nfd("", acc) do
+    acc
+  end
+
+  defp normalize_nfd(<<codepoint :: utf8, rest :: binary>>, acc) when codepoint in 0xAC00..0xD7A3 do
+    {syllable_index, t_count, n_count} = {codepoint - 0xAC00, 28, 588}
+    hangul_l = 0x1100 + div(syllable_index, n_count)
+    hangul_v = 0x1161 + div(rem(syllable_index, n_count), t_count)
+    hangul_t = 0x11A7 + rem(syllable_index, t_count)
+    new =
+      if hangul_t == 0x11A7 do
+        <<hangul_l :: utf8, hangul_v :: utf8>>
+      else
+        <<hangul_l :: utf8, hangul_v :: utf8, hangul_t :: utf8>>
+      end
+    normalize_nfd(rest, acc <> new)
```

Comment:
Just to confirm, the hangul above are already in canonical order, right? So we don't need to call canonical order.